website/docs/reference/transforms/rename_fields.md.erb

```diff
@@ -0,0 +1,39 @@
+<%- component = metadata.transforms.rename_fields -%>
+
+<%= component_header(component) %>
+
+## Configuration
+
+<%= component_config_example(component) %>
+
+<%- if !component.requirements.nil? -%>
+## Requirements
+
+<Alert type="danger" fill={true} icon={false}>
+
+<%= component.requirements %>
+
+</Alert>
+
+<%- end -%>
+## Options
+
+<%= fields(component.specific_options_list, heading_depth: 3) %>
+
+<%- if component.env_vars_list.any? -%>
+## Env Vars
+
+<%= fields(component.env_vars_list, heading_depth: 3) %>
+
+<%- end -%>
+<%- if component.output.log -%>
+## Output
+
+<%= component_output(component, component.output.log, heading_depth: 3) %>
+
+<%- end -%>
+## How It Works [[sort]]
+
```

Comment:
I'd add 2 sections here:

1. Nested fields
    Just a simple section that makes it obvious nested fields are supported.

2. Name collissions
    What happens when a field is renamed to an existing field? What if the existing field is an object (nested). Etc.

Reply:
Turns out nested fields weren't working in this yet! So I added them, and added behavioral tests to check things. :)